lib/Counter/stories/Counter.js

```diff
@@ -0,0 +1,32 @@
+import React, { Fragment } from 'react';
+import { storiesOf } from '@storybook/react';
+import { withKnobs, number } from '@storybook/addon-knobs';
+import { ButtonIcon } from 'lib';
+import { Counter } from '../index';
+import StoryItem from '../../../stories/styleguide/StoryItem';
+
+const stories = storiesOf('Components', module);
+stories.addDecorator(withKnobs);
+
+stories.add('Counter', () => {
+  const count = number('Value', 9);
+
+  return (
+    <Fragment>
+      <StoryItem title="Counter on it's own">
+        <Counter>{count}</Counter>
+      </StoryItem>
+      <StoryItem title="Counter on ButtonIcon">
+        <ButtonIcon
+          className="relative group"
+          name="commentsMulti16"
+          size={ButtonIcon.sizes.md}
+        >
+          <Counter className="absolute top-0 right-0 mt mr border-solid border-2 border-white group-hover:border-neutral-95 active:border-blue-90">
```

Comment:
There seems to be quite a lot of classes here. Every time I need to use a counter in a ButtonIcon I'll have to remember to copy this from somewhere.

If we update this too we'll have to make sure we update it everywhere (not a great pattern for "re-usable" components IMO).

I don't think I'd mind if it was just `absolute top-0` but looking at the array of classes we need it feels like it should be a pattern described by the component itself.

---

This feedback also links with your need for tailwind support because `group-active` isn't a thing. But I think you've answered your own question, it isn't a thing because it shouldn't be.

If you want to style things based on an active state then move that thing into the component and give it a place there. There are plenty of ways to do this but maybe;

- add a prop to `ButtonIcon` ... this can be `Counter` and then you can decorate that with classes when the button is active. If I've misunderstood your active question, let me know 👍 (I'm assuming not though cause you wouldn't want to do anything with group active pseudo, right?)

Reply:
I agree this is a lot of classes to remember to put in for each button icon we want to use this with. This could be an opportunity for a tailwind component https://tailwindcss.com/docs/extracting-components/ 

As for group active we could use https://github.com/benface/tailwindcss-interaction-variants

Reply:
Kyle and I have discussed this further. I was trying to avoid the trap of coupling `Counter` and `ButtonIcon` with a prop. Kyle had the great idea of making this a compound pattern. `Counter` still exists on its own, but there is a variant called `ButtonIcon.Counter` which brings with it all of the specific styles to render that button at the top right.

Reply:
UPDATE:

I hadn't seen Amees comment when I was replying to this!

I have ended up doing a lot of this in CSS. As stated above I started going down the compound component route, but all that compound component was doing was adding some extra styles, which I could just do in CSS. I started thinking that way and a lot of my other issues started becoming easier to fix. I could apply active styling to the counter now!